extension/src/background/websocket-client.ts

```diff
@@ -4,6 +4,7 @@ export default class WebSocketClient {
 
     public constructor(host: string) {
         this.host = host;
+        this.webSocket = new WebSocket(this.host);
```

Comment:
Alors je l'avais enlevé d'ici, car si ta websocket est fermée, il me semble que tu es obligé d'en instancier une nouvelle pour lancer une nouvelle connexion, et du coup c'était assez relou car tu devais relancer l'application WWM. 
Par ailleurs, vu que ce code est évalué dès le chargement du background script, ça veut dire que même si tu te sers pas de l'application WWM, tu va instancier des WS sur des ports

(enfin corrige moi si j'ai mal compris, mais de ce que j'avais vu dans le navigateur, la WS essayait tout le temps de se connecter dès qu'elle était instanciée)

SI tu veux garder cette implémentation faudrait aller changer ligne 51 "case WebSocket.CLOSED" aussi

Reply:
I moved this in the constructor because the `ensureOpened()` method began by:
```javascript
if (!this.webSocket) {
   connect()
}
```

=> With that block, if I want to test `ensureOpened()` correctly, `this.webSocket` is never defined, so the different cases are never reached. Plus `webSocket` is a private property, so I can't mock it.

What I can do, is move this block to a separate method, which I will have to make public, if I want to alter its behaviour.
Or, as you say: I can change the CLOSED case

Reply:
@Startouf What do you prefer, between these 2 alternatives ? I'm really undecided

Reply:
Both are fine, but please make sure the extension works when joining a room multiple times without having to manually restart the extension. ie:
- Start the dev server, open the extension background script console log
- On a youtube video, open the extension, click "join a room"
- refresh the tab or open a different tab, open the popup and click join the room
- The websocket connection should work both times (it is successfully reset for the second room without crashing)

Reply:
What I ended up doing, is keeping the code as it was before, but calling the getter instead of `this.webSocket` in `ensureOpened`. That way, I can manipulate the WS stub properly

Reply:
What I ended up doing, is keeping the code as it was before, but calling the getter instead of `this.webSocket` in `ensureOpened`. That way, I can manipulate the WS stub properly